modules/rpc/yarpc.go

```diff
@@ -177,13 +184,13 @@ func (c *dispatcherController) mergeConfigs() (conf yarpc.Config, err error) {
 		return conf, errors.New("unable to merge empty configs")
 	}
 
-	conf.Name = c.configs[0].AdvertiseName
```

Comment:
This is not in our power, but YARPC: dispatcher allows us to have only one service name.

Snowflakes can create an extra dispatcher and we expose enough functionality: create and start dispatcher functions, for users to synchronize with the second dispatcher if they'd need it. Anyway, I prefer RPC team to tinker these puzzles.